### PR TITLE
frontend: render newly created game

### PR DESCRIPTION
**Pull request by j2kun**
Blocked by https://github.com/j2kun/lonely-hearts/pull/87

### CODE REVIEW

**Comment by DanielNghiem** [comment]
What should I be able to see after running run.py after checking out this branch?

**Comment by j2kun** [comment]
Create a new room on the website, then copy paste the link into 4 tabs and see the newly created game in the db, and the new hands rendered in the browser for each player

**Comment by DanielNghiem** [comment]
@j2kun I'm checked out to origin/jeremy-frontend.  When I click 'join room' the first time, I get routed to a room page that already has a hand of 6 cards and 4 trick cards displayed.  This display does not change when I copy/paste the room url 4 times, and the displayed cards are the same even when I restart the database or the app.   What is going on here?

When I inspect the page I see an ERR_CONNECTION_REFUSED error pop up every few seconds:

http://0.0.0.0:8080/socket.io/?EIO=3&transport=polling&t=Llfg7V3 net::ERR_CONNECTION_REFUSED

**Comment by j2kun** [comment]
So what you see initially is left over from when I was just messing around. You can ignore that for now, and the `render_waiting_for_players` function will handle what to do before the game is started later. 

It looks to me like you need to set `HOST` and `PORT` to point to your cloud9 url. I don't remember what yours was, but mine was something like `HOST=lonely-hearts-j2kun.c9users.io` with `PORT=8080`

**Comment by DanielNghiem** [comment]
My .env:
```
HOST=0.0.0.0
WWW_HOST=hearts-dnghiem.c9users.io
PORT=8080

DATABASE_URL=mongodb://localhost:27017/hearts
DEBUG=True
```

**Comment by j2kun** [comment]
Set `HOST=hearts-dnghiem.c9users.io` as well.

**Comment by DanielNghiem** [comment]
Setting HOST=hearts-dnghiem.c9users.io produced:
```
dnghiem:~/workspace/lonely-hearts ((c392b6f...)) $ python run.py 
 * Restarting with stat
 * Debugger is active!
 * Debugger pin code: 279-131-420
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/workspace/lonely-hearts/venv/lib/python3.5/site-packages/flask_socketio/__init__.py", line 457, in run_server
    eventlet_socket = eventlet.listen((host, port))
  File "/home/ubuntu/workspace/lonely-hearts/venv/lib/python3.5/site-packages/eventlet/convenience.py", line 43, in listen
    sock.bind(addr)
OSError: [Errno 99] Cannot assign requested address
```

File: hearts/static/room.js [hunk 1]
```diff
@@ -7,15 +7,35 @@ function apiCard(displayCard) {
...
+            'hands': {str: [str]},
+            'tricks': [[str]],
+            'hearts': self.hearts_broken,
+            'current_scores': {str: int},
+            'final_scores': {str: int},
+            'is_over': boolean,
+        }, ...
+    ],
+    'round_number': int,
+    'scores': {str: int},
```
Comment (DanielNghiem) [inline hearts/static/room.js]: According to hearts/game/hearts.py, `'scores'` is a list of scores from each completed round.

So the line should be 
`'scores': [{str: int}],

File: hearts/static/room.js [hunk 2]
```diff
@@ -7,15 +7,35 @@ function apiCard(displayCard) {
...
+{
+    'players': [str],
+    'max_points': int,
+    'rounds': [
+        {
+            'players': str,
+            'direction': str,
+            'turn': int,
+            'hands': {str: [str]},
+            'tricks': [[str]],
```
Comment (DanielNghiem) [inline hearts/static/room.js]: Currently, a serialized trick is in the form `{str: {int: str}}`  (cf. https://github.com/j2kun/lonely-hearts/blob/master/hearts/game/hearts.py#L545-L554)

I don't remember why I made it that way.  A trick really should simply be a list of tuples `[(player_string, card_string)]`  Is this worth refactoring?

**Comment by DanielNghiem** [comment]
Other than my two comments on https://github.com/j2kun/lonely-hearts/pull/85/files/b3c1905aec624c60a0c0f2c0f3939f05218c1581..baaa73a28d6890f4fcd30cfd36b552380e0b9346#diff-1612638374fbedffcb2f2a6532c63795, it looks good to me.

File: hearts/static/room.js (hunk 2 above)
Comment (j2kun) [inline hearts/static/room.js]: I'm gonna merge this as is, but yeah if you think so then let's refactor.

**Pull request merged**